python/Ganga/GPIDev/Lib/Job/Job.py

```diff
@@ -1758,6 +1758,8 @@ def doit(f):
                         self.application.decrementShareCounter(self.application.is_prepared.name)
                         for _ in self.subjobs:
                             self.application.decrementShareCounter(self.application.is_prepared.name)
+                    if self.application.getShareCounterVal(self.application.is_prepared.name)==0:
```

Comment:
Yes I agree but the shareRef and the ShareDir objects are somewhat decoupled. I'll try and rewrite it to be more as one would expect.